<a href="https://colab.research.google.com/github/Umizmar/The_theory_of_probability/blob/main/%D0%A3%D1%80%D0%BE%D0%BA_3_EDA_(exploratory_data_analysis)_%D0%B8%D0%BB%D0%B8_%D0%A0%D0%B0%D0%B7%D0%B2%D0%B5%D0%B4%D0%BE%D1%87%D0%BD%D1%8B%D0%B9_%D0%B0%D0%BD%D0%B0%D0%BB%D0%B8%D0%B7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Теория вероятностей и математическая статистика

###Урок 3. EDA (exploratory data analysis) или Разведочный анализ

1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [ ]:
salary_list = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]

In [ ]:
a_mean =  sum(salary_list)/len(salary_list)
print ('Среднее арифметическое -', a_mean)

Среднее арифметическое - 65.3


In [ ]:
dev_mean = list(map(lambda x: (x-a_mean)**2, salary_list))
print('Отклонение от среднего в квадрате -', dev_mean)
dispers = sum(dev_mean)/len(dev_mean)
print('Смещенная дисперсия -', dispers)
m_sq_dev = dispers**0.5
print('Среднее квадратичное отклонение -', m_sq_dev)

Отклонение от среднего в квадрате - [1204.0900000000001, 216.0900000000001, 94.09000000000006, 136.89000000000007, 561.6900000000002, 1043.2899999999997, 412.08999999999986, 1624.0899999999997, 0.08999999999999829, 2332.89, 1246.0899999999997, 1705.6899999999998, 68.88999999999996, 106.08999999999995, 22.09000000000003, 94.09000000000006, 0.08999999999999829, 349.6900000000001, 610.0900000000001, 7174.09]
Смещенная дисперсия - 950.11
Среднее квадратичное отклонение - 30.823854398825596


In [ ]:
sdispers = sum(dev_mean)/(len(dev_mean)-1)
print('Несмещенная дисперсия -', sdispers)

Несмещенная дисперсия - 1000.1157894736842


2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Вероятность вытащить 3 белых мяча складывается из 3х событий:  
А1 - из первого ящика вытаскивают 2 белых и из второго 1 белый  
А2- из первого ящика вытаскивают 1 белый и из второго 2 белых  
А3 - из первого ни одного, из второго 3 белых

А = А1+А2+А3

$$P(A1)=\frac{C_5^2}{C_8^2}\times\frac{C_5^1\times{C_7^3}}{C_{12}^4}$$


In [ ]:
from math import factorial as fc

In [ ]:
def prob(n,k):
    return fc(n)/(fc(k)*fc(n-k))

In [ ]:
p1 = prob(5,2)/prob(8,2)*prob(5,1)*prob(7,3)/prob(12,4)
print('А1 =', p1)

А1 = 0.12626262626262627


$$P(A2)=\frac{C_5^1\times C_3^1}{C_8^2}\times\frac{C_5^2\times{C_7^2}}{C_{12}^4}$$

In [ ]:
prob(5,1)

5.0

In [ ]:
p2 = prob(5,1)*prob(3,1)/prob(8,2)*prob(5,2)*prob(7,2)/prob(12,4)
print('А2 =', p2)

А2 = 0.22727272727272724


$$P(A3)=\frac{C_3^2}{C_8^2}\times\frac{C_5^3\times{C_7^1}}{C_{12}^4}$$

In [ ]:
p3 = (prob(3,2)/prob(8,2))*(prob(5,3)*prob(7,1)/prob(12,4))
print(f'А3 = {p3}')

А3 = 0.01515151515151515


А = А1+А2+А3

In [ ]:
f'Овет: {p1+p2+p3:.{4}f}'

'Овет: 0.3687'

3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.


А-в результате выстрела, цель поражена  
В1-выстрел произведен первым спортсменом  
В2-выстрел произведен вторым спортсменом  
В3-выстрел произведен третьим спортсменом.

P(B1)=P(B2)=P(B3)=1/3

Вероятность наступления события A: $$P(A)=P(B_1)\cdot P_{B_1}(A)+P(B_2)\cdot P_{B_2}(A)+P(B_3)\cdot P_{B_3}(A)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$$

In [2]:
PA=1/3*0.9+1/3*0.8+1/3*0.6
PA

0.7666666666666666

Далее воспользуемся формулой Байеса, и найдем вероятность события $B_n$
$$P_A(B_n)=\frac{P(B_n)\cdot P_{B_n}(A)}{P(A)}\Rightarrow {;}P_A(B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}{;}P_A(B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}{;}P_A(B_1)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$$

In [5]:
PAB1=1/3*0.9/PA
PAB2=1/3*0.8/PA
PAB3=1/3*0.6/PA
print(f'Первый спортсмен-{PAB1:.4f}, второй спортсмен-{PAB2:.4f}, третий спортсмен-{PAB3:.4f}')

Первый спортсмен-0.3913, второй спортсмен-0.3478, третий спортсмен-0.2609


4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?


Студентов на факультетах по условию:  
A=B=1/4  
c=1/2  
D - событие-сессия сдана  
Полная вероятность события D:$$\frac{1}{4}\cdot 0.8+\frac{1}{4}\cdot 0.7+\frac{1}{2}\cdot 0.9$$

In [6]:
D = 1/4*0.8+1/4*0.7+1/2*0.9
D

0.825

По формуле Байеса вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию

In [8]:
A=1/4*0.8/D
B=1/4*0.7/D
C=1/2*0.9/D
print(f'Факультет А- {A:.4f}, Факультет B- {B:.4f}, Факультет C- {C:.4f}')

Факультет А- 0.2424, Факультет B- 0.2121, Факультет C- 0.5455


5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

Введем следующие события:  
$A1$ - выход из строя первой детали. $P(A1)=0.1$  
$\bar A1$ - событие противоположное $A1$. $P(\bar A1)=0.9$  
$A2$-второй детали.$P(A2)=0.2$. $P(\bar A2)=0.8$  
$A3$-третьей детали.$P(A2)=0.25$. $P(\bar A2)=0.75$

a) Вышли из строя все детали
$$P(A_{123})=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$$

In [13]:
A123 = 0.1*0.2*0.25
print(f'{A123:.4f}')

0.0050


б) Чтобы произошло событие "вышли из строя 2 детали", необходимо что бы случилось одна из следующих комбинаций событий:
1. Произошли события $A_1$ **и** $A_2$ **и** $\bar A_3$;  
2. Произошли события $A_1$ **и** $A_3$ **и** $\bar A_2$;  
3. Произошли события $A_2$ **и** $A_3$ **и** $\bar A_1$.  

А вероятностью этого события будет сумма вероятностей этих комбинаций  
    $$P(2)=P(A_1)\\cdot P(A_2)\\cdot P(\\bar A_3)+P(A_1)\\cdot P(A_3)\\cdot P(\\bar A_2)+P(A_2)\\cdot P(A_3)\\cdot P(\\bar A_1)=0.1\\cdot 0.2\\cdot 0.75+0.1\\cdot 0.25\\cdot 0.8+0.2\\cdot 0.25\\cdot 0.9$$

In [14]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Р(2) = {P2: .4f}')

Р(2) =  0.0800


в) Событие "выйдет из строя хотя бы одна деталь" можно рассмотреть как противоположное событию "не вышло из строя ни одной детали".
$P(\bar A_{123})=P(\bar A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)=0.9*0.8*0.75$

In [15]:
PA123=0.9*0.8*0.75
print(f'{PA123: .4f}')

 0.5400


Вероятность события "выйдет из строя хотя бы одна деталь" = $1-P(\bar A_{123})$

In [17]:
P=1-PA123
print(f'{P: .4f}')

 0.4600


г) Событие "из строя выйдут от одной до двух деталей" произойдет при наступлении события "из строя выйдет одна деталь" или "из строя выйдут 2 детали", следовательно его вероятность будет равна сумме вероятностей этих событий: $P=P(1)+P(2)$  
    найдем $P(1)$

Для того, чтобы произошло событие "из строя выйдет одна деталь", необходимо что бы случилось одна из следующих комбинаций событий:
1. Произошли события $A_1$ **и** $\bar A_2$ **и** $\bar A_3$
2. Произошли события $A_2$ **и** $\bar A_1$ **и** $\bar A_3$  
3. Произошли события $A_3$ **и** $\bar A_1$ **и** $\bar A_2$.



А вероятностью этого события  $P(1)$  будет суммой вероятностей этих комбинаций событий
$$P(1)=P(A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)+P(A_2)\cdot P(\bar A_1)\cdot P(\bar A_3)+P(A_3)\cdot P(\bar A_1)\cdot P(\bar A_2)=0.1\cdot 0.8\cdot 0.75+0.2\cdot 0.9\cdot 0.75+0.25\cdot 0.9\cdot 0.8$$

In [18]:
P1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'{P1: .4f}')

 0.3750


$P=P(1)+P(2)=0.375+0.08$

In [19]:
P = P1 + P2
P

0.455